scripts/cli.rb

```diff
@@ -46,11 +49,19 @@ def parse(args = ARGV)
         options[:only] = Target.supported_types
       end
 
-      Trollop.die(:reference, git_ref_desc) if options[:reference].to_s.empty?
-      options[:reference] = options[:reference].to_s.strip
+      # --reference overrides all other reference arguments
+      unless options[:reference].to_s.empty?
+        options[:manageiq_ref]  = options[:reference]
+        options[:appliance_ref] = options[:reference]
+        options[:build_ref]     = options[:reference]
+        options[:ssui_ref]      = options[:reference]
+      end
+
+      Trollop.die(:manageiq_ref, git_ref_desc) if options[:manageiq_ref].to_s.empty?
+      options[:manageiq_ref] = options[:manageiq_ref].to_s.strip
```

Comment:
I think you want to move this to 53.5 - before using reference to set all the other references